avogadro/core/molecule.h

```diff
@@ -680,6 +710,31 @@ inline bool Molecule::setFormalCharge(Index atomId, signed char charge)
   return false;
 }
 
+inline Vector3ub Molecule::color(Index atomId) const
+{
+  return atomId < m_colors.size() ? m_colors[atomId] : Vector3ub(0, 0, 0);
```

Comment:
Here is where I think we make the change to look up the standard color for the atomic number if the array is too short. This would let us color by atomic number as we have by default, and change in custom colors lazily for people who want to.